Abhi-AnimeX/app/src/main/java/com/abhijith/animex/ui/screens/animedetails/AnimeDetails.kt

```diff
@@ -33,18 +32,36 @@ import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.text.font.Font
 import androidx.compose.ui.text.font.FontFamily
 import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.unit.dp
+import androidx.core.content.ContextCompat.getString
+import androidx.lifecycle.viewmodel.compose.viewModel
 import coil.compose.AsyncImage
 import coil.request.ImageRequest
 import com.abhijith.animex.R
+import com.abhijith.animex.domain.model.AnimeItem
 import com.abhijith.animex.ui.components.GenreTag
+import com.abhijith.animex.ui.components.RatingTag
 import com.abhijith.animex.ui.components.StatItem
+import com.abhijith.animex.ui.components.YoutubeVideoTile
+import com.abhijith.animex.ui.screens.animedetails.viewmodel.AnimeDetailsViewModel
+import com.abhijith.animex.ui.screens.error.ErrorScreen
+import com.abhijith.animex.ui.screens.loading.LoadingScreen
 
 @Composable
-fun AnimeDetails() {
-    val genreList = listOf("comedy", "gag humor", "school", "adventure")
+fun AnimeDetails(animeDetailsViewModel: AnimeDetailsViewModel = viewModel()) {
+    when (val uiState = animeDetailsViewModel.uiState.collectAsState().value) {
+        is AnimeDetailsUiState.Error -> ErrorScreen(uiState.message)
+        is AnimeDetailsUiState.Loading -> LoadingScreen()
+        is AnimeDetailsUiState.Success -> AnimeDetailsInfo(uiState.item, animeDetailsViewModel)
+    }
+}
 
+@Composable
+fun AnimeDetailsInfo(animeItem: AnimeItem, animeDetailsViewModel: AnimeDetailsViewModel) {
+    val yearAndJapaneseName =
+        animeDetailsViewModel.formatYearAndJapaneseName(animeItem.year, animeItem.japaneseName)
```

Comment:
why coming here and then doig the formating, why not in data or domain layer?

Reply:
moved out  of here and added to domain class.. @anandwana001